code/src/hooks/use-filters-with-apply/parse-serialized-set/parseSerializedSet.ts

```diff
@@ -1,23 +1,27 @@
 import deserializeFromQueryString from "@/hooks/use-filters-with-apply/deserialize-from-query-string/deserializeFromQueryString";
 
-const parseSerializedSet = (value: string) => {
-  const match = value.match(/^{((?:\w+(?:,\w+)*)?)}$/);
+// matches list of values, separated by comma and wrapped into brackets
+// Examples: {1,2,3}, {delivered,completed}, {}, {value}
+const serializedSetPattern = /^{((?:\w+(?:,\w+)*)?)}$/;
+
+const parseSerializedSet = <Value>(value: string) => {
+  const match = value.match(serializedSetPattern);
 
   const arrayMatch = match?.[1];
 
-  if (arrayMatch === "") {
-    return new Set();
+  if (arrayMatch === undefined) {
+    return null;
   }
```

Comment:
I have suggestion
```suggestion
if (!arrayMatch) return null
```

Reply:
Thanks for suggestion! Maybe it is not required here. I like to have brackets because we can add more content to if block in less steps and for me it looks more readable. But if you insist I can apply your suggestion

Reply:
> Thanks for suggestion! Maybe it is not required here. I like to have brackets because we can add more content to if block in less steps and for me it looks more readable. But if you insist I can apply your suggestion

If you want with `{}` ok.
But replace check with `!arrayMatch` compare with `undefined` it's not a good solution

Reply:
Sorry, I can't do it. I hope you know how [`match()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match) works. For me it has strange interface, but matching against regex is very powerfull.

I want to explain why i did it like it is. `match` works so that if there is no match, it returns empty matched array, so `arrayMatch` will always be undefined if we try to access `array[1]` when `array` is empty. If there is match, array of matches is returned. Match are things wrapped inside brackets. For example `(match1, (match2, (match3)))` with return array of `[(match1, (match2, (match3))), (match2, (match3)), (match3)]` if there is a match against regex. We also can ignore matching some group using non-capturing group operator, like i did in regex above. [Read more non capturing group](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Non-capturing_group)

In my specific regex i can parse empty set from string `'{}'`. The second match in my case will be emty string, so i can check it below and return empty set. Thats why I can't use `!arrayMatch` because as you know it accepts all falsy values, in my case both `""` and `undefined`, but it is not what I want here